subprojects/language-java/src/main/java/org/gradle/api/tasks/compile/JavaCompile.java

```diff
@@ -75,10 +76,12 @@
  */
 @CacheableTask
 public class JavaCompile extends AbstractCompile {
-    private final CompileOptions compileOptions = new CompileOptions();
+    private final CompileOptions compileOptions;
     private JavaToolChain toolChain;
 
     public JavaCompile() {
+        CompileOptions compileOptions = getServices().get(ObjectFactory.class).newInstance(CompileOptions.class);
```

Comment:
Nit: I would just inline compileOptions.

Reply:
My pet peeve is not to mutate fields and use their values in the same method if I can avoid it. Unless you feel strongly, I'd leave this as it is.

Reply:
Looks like I am so used to some patterns that I don't even notice that there is less immutability in your solution.

Reply:
How's there less?

Reply:
Sorry, less -> more.